rest/user_api_test.go

```diff
@@ -262,23 +264,28 @@ func TestUsersAPIDetailsWithLimit(t *testing.T) {
 func TestUserAPI(t *testing.T) {
 
 	// PUT a user
-	rt := NewRestTesterDefaultCollection(t, nil) // CBG-2618: fix collection channel access
+	rt := NewRestTester(t, nil)
 	defer rt.Close()
 	ctx := rt.Context()
+	collection := rt.GetSingleTestDatabaseCollection()
+	c := collection.Name()
+	s := collection.ScopeName()
 
 	RequireStatus(t, rt.SendAdminRequest("GET", "/db/_user/snej", ""), 404)
-	response := rt.SendAdminRequest("PUT", "/db/_user/snej", `{"email":"jens@couchbase.com", "password":"letmein", "admin_channels":["foo", "bar"]}`)
+	response := rt.SendAdminRequest("PUT", "/db/_user/snej", `{"email":"jens@couchbase.com", "password":"letmein", `+AdminChannelGrant(collection, `"admin_channels":["foo", "bar"]`)+`}`)
 	RequireStatus(t, response, 201)
 
+	user, err := rt.ServerContext().Database(ctx, "db").Authenticator(ctx).GetUser("snej")
+	require.NoError(t, err)
 	// GET the user and make sure the result is OK
 	response = rt.SendAdminRequest("GET", "/db/_user/snej", "")
 	RequireStatus(t, response, 200)
 	var body db.Body
 	require.NoError(t, base.JSONUnmarshal(response.Body.Bytes(), &body))
 	assert.Equal(t, "snej", body["name"])
 	assert.Equal(t, "jens@couchbase.com", body["email"])
-	assert.Equal(t, []interface{}{"bar", "foo"}, body["admin_channels"])
-	assert.Equal(t, []interface{}{"!", "bar", "foo"}, body["all_channels"])
+	assert.Contains(t, "bar:1,foo:1", user.CollectionExplicitChannels(s, c).String())
```

Comment:
What required the change to Contains and string matching here?  It feels less robust.